03_letters/alextmz/main.go

```diff
@@ -0,0 +1,35 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"sort"
+	"strconv"
+	"strings"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	fmt.Fprintln(out, strings.Join(sortLetters(letters("aba")), "\n"))
+}
+
+func letters(s string) map[rune]int {
+	m := make(map[rune]int)
+	for _, v := range s {
+		m[v]++
+	}
+	return m
+}
+
+func sortLetters(m map[rune]int) []string {
+	s := make([]string, len(m))
+	i := 0
+	for k, v := range m {
+		s[i] = string(k) + ":" + strconv.Itoa(v)
```

Comment:
No action needed:
this is ok, but `fmt.Sprintf("%c:%d",k,v)` seems to be cleaner.

Reply:
Yep, it is. Bad habit because Sprintf is in my mind a 'slow and heavy' function, but I wonder if that holds true for Go too.

Reply:
yes it's still true :cry: `Sprintf` seems to be 2.5  times slower

```
func fSprintf(i int, c rune) string {
	return fmt.Sprintf("%c:%d", c, i)
}
func fConcat(i int, c rune) string {
	return string(c) + ":" + strconv.Itoa(i)
}
```

```bash
 n0npax@narrenturm  /tmp/aa   go test -bench=. 
goos: linux
goarch: amd64
BenchmarkSprintf-8   	10000000	       134 ns/op
BenchmarkConcat-8    	30000000	        53.2 ns/op
PASS
ok  	_/tmp/aa	3.151s
```

Reply:
Man you're a Nerd with capital N (that's a compliment btw). You didn't need to go and measure it! But as you did, thanks! Now I know how to quickly bench those :)

Reply:
Thanks :) I was just curious what is the performance difference between those two.